src/app/api/dashboard/[type]/route.js

```diff
@@ -165,6 +181,11 @@ export async function DELETE(req, { params }) {
   try {
     if (types.has(params.type)) {
       objects.map(async (object) => {
+        const snapshot = await getDoc(doc(db, "users", object));
```

Comment:
the object itself should have the role as well so we dont need to make an API call for it